### PR TITLE
feat: azurem bump to 3.39.x (#8)

### PR DESCRIPTION
**Pull request by jmolnar-zscaler**
* refactor resources for 3.x provider: azurerm_public_ip, azurerm_lb_probe, and cc_lb_rule
* Fix for Azure Load Balancer frontend_ip creation in Regions that do not support availability zones

### CODE REVIEW

**Comment by jmolnar-zscaler** [comment]
fix for bug-134428 already verified by QA. Need a changelog and release version bump as well.

File: modules/terraform-zscc-lb-azure/main.tf [hunk 1]
```diff
@@ -43,12 +44,11 @@ resource "azurerm_lb_backend_address_pool" "cc_lb_backend_pool" {
 # Define load balancer health probe parameters
 ################################################################################
 resource "azurerm_lb_probe" "cc_lb_probe" {
-  name                = "${var.name_prefix}-cc-lb-probe-${var.resource_tag}"
-  resource_group_name = var.resource_group
```
Comment (vkrishnamurthy-zscaler) [inline modules/terraform-zscc-lb-azure/main.tf:47]: Was it complaining when you had resouce_group_name?  This is not applicable I believe and hence you removed it.  How did you figure this out?

File: modules/terraform-zscc-lb-azure/main.tf [hunk 2]
```diff
@@ -43,12 +44,11 @@ resource "azurerm_lb_backend_address_pool" "cc_lb_backend_pool" {
 # Define load balancer health probe parameters
 ################################################################################
 resource "azurerm_lb_probe" "cc_lb_probe" {
-  name                = "${var.name_prefix}-cc-lb-probe-${var.resource_tag}"
-  resource_group_name = var.resource_group
-  loadbalancer_id     = azurerm_lb.cc_lb.id
-  protocol            = "Http"
-  port                = var.http_probe_port
-  request_path        = "/?cchealth"
+  name            = "${var.name_prefix}-cc-lb-probe-${var.resource_tag}"
```
Comment (vkrishnamurthy-zscaler) [inline modules/terraform-zscc-lb-azure/main.tf:47]: Even though there is no change, why is github marking them as differences?

**Review by vkrishnamurthy-zscaler — COMMENTED** [review]
Please see comments.  Looks good otherwise.

**Comment by jmolnar-zscaler** [comment]
@smone77 @vkrishnamurthy-zscaler can I get a thumbs up here please?

**Review by vkrishnamurthy-zscaler — APPROVED** [review]
looks good

**Review by vkrishnamurthy-zscaler — APPROVED** [review]
Looks Goood

**Pull request merged**